components/BookmarkIcon.js

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import styled from "styled-components";
+import { FaBookmark, FaRegBookmark } from "react-icons/fa";
+
+const StyledBookmarkIcon = styled.button`
+  border: none;
+  cursor: pointer;
+  background-color: transparent;
+`;
+
+export function BookmarkIcon({
+  onHandleBookmarkedIcon,
+  recipe,
+  bookmarkedRecipesIDs,
+}) {
+  function handleClick() {
+    onHandleBookmarkedIcon(recipe);
+  }
+
+  const isBookmarked = bookmarkedRecipesIDs.includes(recipe.id);
+
+  return (
+    <StyledBookmarkIcon onClick={handleClick}>
+      {isBookmarked ? <FaBookmark /> : <FaRegBookmark />}
+    </StyledBookmarkIcon>
```

Comment:
this is quite an "intelligent" component. you might want to consider refactoring the code to reduce the props you have to pass down: with two props "onClick" and "isBookmarked" you can remove the handleClick and isBookmarked checking logic from this component and move it higher up. 
It is completely up to you if you want to change this. The larger the apps become, the better is it to keep the number of props small and to cluster the component logic instead of speading them into many small components.